systrad/domain/indicator/quote_extrenum_model.py

```diff
@@ -0,0 +1,93 @@
+"""Quote Extrenum Model.
+
+将来の一定期間における銘柄の最高値/最安値を予測するモデル
+気になる点：データフレームセットの仕方、X_FEATUREの指定方法、indicator_interactorでのstock_dfの更新方法
+TODO：x_featureの決定方法、推論結果をstock_dfにどう反映するか
+
+"""
+import pickle
+
+import xgboost as xgb
+
+from systrad.domain.indicator.model import ModelABC
+from systrad.domain.indicator.stock import StockDF
+
+
+class Xgboost(ModelABC):
+    """Xgboost
+
+    Xgboostによる予測モデル
+
+    Attributes:
+        original_feature_columns (frozenset): 未処理状態での特徴量カラム
+        extracted_feature_columns (frozenset): 抽出工程により追加されるカラム
+                                          Stock内でConfigに従って定義する
+        __df (pd.DataFrame): 内部保持するデータフレーム
+
+    """
+    X_TARGET_DAYS = [1, 3, 5, 10, 20, 40, 60, 80, 120]
+    X_FEATURE = ["EndOfDayQuote Date", "EndOfDayQuote Close"] + ["return_" + str(d) for d in TARGET_DAYS] + [
+        "art_" + str(d) for d in TARGET_DAYS
+    ] + ["art_b_" + str(d)
+         for d in TARGET_DAYS] + ["art_s_" + str(d)
+                                  for d in TARGET_DAYS] + ["ma_gap_" + str(d) for d in TARGET_DAYS]
+    Y_FEATURE = ["label_high_20", "label_low_20"]
+    model_high_path = "hoge"
+    model_low_path = "hoge"
+
+    def set_data(self, stock_df: pd.DataFrame) -> None:
+        """StockのDataFrameの読み込み
+
+        学習、推論に必要なDataFrameを読み込む
+
+        Args:
+            stock_df (pd.DataFrame): データフレームの読み込み
+        """
+        self.__df = stock_df.copy()
+        self.__result = {}
+
+    def set_model(self) -> None:
+        """学習済みモデルの取得
+
+        学習済みモデルをロードする
+        
+        """
+        self.model_high = pickle.load(open(self.model_high_path, "rb"))
+        self.model_low = pickle.load(open(self.model_low_path, "rb"))
+
+    def pre_process(self) -> None:
+        """前処理"""
+        self.__df = self.__df.replace([np.inf, -np.inf], np.nan)
+        self.__df = self.__df.fillna(0)
+
+    def post_process() -> None:
+        """後処理"""
+        pass
+
+    def save_model() -> None:
+        """学習済みモデルの保存"""
+        pass
+
+    def train_and_savemodel() -> None:
+        """
+        学習とモデルの保存
+        """
+        pass
+
+    def predict(self) -> None:
```

Comment:
[imo]
```suggestion
    def predict(self, stock_df: StockDF) -> pd.Seleries:
```

上でも書いたけど、こっちの方が私は好き